src/components/safe-apps/AppFrame/index.tsx

```diff
@@ -117,6 +132,31 @@ const AppFrame = ({ appUrl, allowedFeaturesList }: AppFrameProps): ReactElement
         blockExplorerUriTemplate,
       }
     },
+    onSetSafeSettings: (settings: SafeSettings) => {
+      const EIP_1271_SUPPORTED_SAFE_VERSION = '1.0.0'
+
+      if (typeof settings.offChainSigning === 'boolean' && gte(safe.version, EIP_1271_SUPPORTED_SAFE_VERSION)) {
+        setOffChainSigning(settings.offChainSigning)
+      }
+
+      return true
```

Comment:
Perhaps we should return the new settings instead always true to the caller. wdyt?

Reply:
Good point. I've adjusted it and will update the SDK accordingly.